packaging/docker/Dockerfile

```diff
@@ -39,11 +39,13 @@ FROM netdata/base:${ARCH}
 
 # Conditional subscribiton to Polyverse's Polymorphic Linux repositories
 RUN if [ "$(uname -m)" == "x86_64" ]; then \
+        apk update && apk upgrade; \
```

Comment:
Why `apk update` and `apk upgrade` now is before polyverse at line 42 instead of after?

Reply:
It should be both - before and after. We learned a lot during this testing/repro and using some best practices.

So let's say we want Polymorphic `curl`. If the base container is not updated, then this can happen:

`curl 1.0` -> `polymorphic curl 2.0`

So we're upgrading AND reinstalling all at once. This had a few risks and failure modes we took the opportunity to fix. Instead now what happens is:

`curl 1.0` -> `curl 2.0` (upgrade)
`curl 2.0` -> `polymorphic curl 2.0` (reinstall)

The first upgrade allows any triggers, upgrade scripts, config migrations, etc. to happen cleanly without Polyverse getting in the way.

Reply:
That's an excellent question @paulkatsoulakis. You're absolutely correct that the first upgrade is meant to ensure you're on the latest versions, before we install polymorphic versions.

On your question about switching packages - that is the one thing we guarantee and ensure will always work - our packages show up as drop-in replacements for any upstream packages. Versions, dependencies, etc. The switch is transparent and easy. In fact you see it now - when we removed py2-psycopg2 from our repo, nothing changed on the image, but the error went away.

Reply:
cool thanks